### PR TITLE
Do not overwrite the language when it's just missing for one app

### LINKED ISSUE #20666: Can't change languages (OC 8.2.x)
Hi!
I'm having troubles changing languages on OC 8.2.0 and 8.2.1.
I can select the language I want, but it's applied only when spanish or english is selected. Otherwise, the last one used is applied.
In the database the language is changed correctly.

I'm using Firefox and Chrome, and both had the issue.
I have tried a clean install (installing OC on a new folder and creating a new database), with the same results.

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #20666
### Steps
1. Make sure your browsers "accept language header" does not contain Russian
2. Make sure you have no default language in the config.php
3. Select Russian as language on the personal page
4. Visit the activity page
### Expected

The page should be in Russian
### Actual

It's in the first language of your accept header

@karlitschek should backport to 8.2

@MorrisJobke @PVince81 please review

### CODE REVIEW

File: lib/private/l10n.php [hunk 1]
```diff
@@ -116,13 +116,17 @@ public static function setLanguageFromRequest() {
 				$preferred_language = str_replace('-', '_', $preferred_language);
 				foreach ($available as $available_language) {
 					if ($preferred_language === strtolower($available_language)) {
-						self::$language = $available_language;
+						if (!self::$language) {
```
Comment (PVince81) [inline lib/private/l10n.php]: Should we move the if outside the loop ? If `$language` is already set we might not need to to into the loop at all ?

Comment (nickvergessen) [inline lib/private/l10n.php]: well we return after we reached the if anyway, so it's a one time check

Comment (PVince81) [inline lib/private/l10n.php]: fair enough

**Comment by PVince81** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by MorrisJobke** [comment]
@karlitschek Can you comment on the backport request? The ticket was scheduled for 8.2.2 ;)

**Comment by DeepDiver1975** [comment]
@nickvergessen please open backport pr - THX

**Comment by karlitschek** [comment]
Agreed. Please backport.

**Pull request merged**